src/nodes.jl

```diff
@@ -259,11 +259,20 @@ struct XMLElementIter
     parent_ptr::Xptr
 end
 
-Base.start(it::XMLElementIter) =
-    ccall((:xmlFirstElementChild,libxml2), Xptr, (Xptr,), it.parent_ptr)
-Base.done(it::XMLElementIter, p::Xptr) = (p == C_NULL)
-Base.next(it::XMLElementIter, p::Xptr) =
-    (XMLElement(p), ccall((:xmlNextElementSibling,libxml2), Xptr, (Xptr,), p))
+if isdefined(Base, :iterate)
+    function Base.iterate(it::LightXML.XMLElementIter,
+            state=(ccall((:xmlFirstElementChild,LightXML.libxml2), LightXML.Xptr, (LightXML.Xptr,), it.parent_ptr), 0))
+        state[1] == C_NULL && return nothing
+        el = LightXML.XMLElement(state[1])
+        (el, (ccall((:xmlNextElementSibling,LightXML.libxml2), LightXML.Xptr, (LightXML.Xptr,), state[1]), 0))
```

Comment:
don't need `LightXML.` qualification anywhere here, we're inside the package module

Reply:
Nice Catch.  Thanks!  I removed the not-needed qualifications.